# Somatic


## Inputs


### Required inputs
<p name="Somatic.bwaIndex">
        <b>Somatic.bwaIndex</b><br />
        <i>struct(fastaFile : File, indexFiles : Array[File]) &mdash; Default: None</i><br />
        The BWA index files.
</p>
<p name="Somatic.dbsnpVCF">
        <b>Somatic.dbsnpVCF</b><br />
        <i>File &mdash; Default: None</i><br />
        dbsnp VCF file used for checking known sites.
</p>
<p name="Somatic.dbsnpVCFIndex">
        <b>Somatic.dbsnpVCFIndex</b><br />
        <i>File &mdash; Default: None</i><br />
        Index (.tbi) file for the dbsnp VCF.
</p>
<p name="Somatic.dockerImagesFile">
        <b>Somatic.dockerImagesFile</b><br />
        <i>File &mdash; Default: None</i><br />
        A YAML file describing the docker image used for the tasks. The dockerImages.yml provided with the pipeline is recommended.
</p>
<p name="Somatic.referenceFasta">
        <b>Somatic.referenceFasta</b><br />
        <i>File &mdash; Default: None</i><br />
        The reference fasta file.
</p>
<p name="Somatic.referenceFastaDict">
        <b>Somatic.referenceFastaDict</b><br />
        <i>File &mdash; Default: None</i><br />
        Sequence dictionary (.dict) file of the reference.
</p>
<p name="Somatic.referenceFastaFai">
        <b>Somatic.referenceFastaFai</b><br />
        <i>File &mdash; Default: None</i><br />
        Fasta index (.fai) file of the reference.
</p>
<p name="Somatic.sampleConfigFile">
        <b>Somatic.sampleConfigFile</b><br />
        <i>File &mdash; Default: None</i><br />
        The samplesheet, including sample ids, library ids, readgroup ids and fastq file locations.
</p>

### Other common inputs
<p name="Somatic.adapterForward">
        <b>Somatic.adapterForward</b><br />
        <i>String? &mdash; Default: "AGATCGGAAGAG"</i><br />
        The adapter to be removed from the reads first or single end reads.
</p>
<p name="Somatic.adapterReverse">
        <b>Somatic.adapterReverse</b><br />
        <i>String? &mdash; Default: "AGATCGGAAGAG"</i><br />
        The adapter to be removed from the reads second end reads.
</p>
<p name="Somatic.cnvPanelOfNormals">
        <b>Somatic.cnvPanelOfNormals</b><br />
        <i>File? &mdash; Default: None</i><br />
        The panel of normals file to be used for CNV calling. If not provided (and performCnvCalling is set to true) then this will be generated on the fly using the samples lacking a control sample in the samplesheet.
</p>
<p name="Somatic.CNVs.annotatedIntervals">
        <b>Somatic.CNVs.annotatedIntervals</b><br />
        <i>File? &mdash; Default: None</i><br />
        An annotated set of intervals as generated by AnnotateIntervals (eg. using CNV-PON.wdl). Will be ignored if PON is provided.
</p>
<p name="Somatic.generateCnvPanelOfNormals.annotateIntervals.mappabilityTrack">
        <b>Somatic.generateCnvPanelOfNormals.annotateIntervals.mappabilityTrack</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to gatk AnnotateIntervals' `--mappability-track` option.
</p>
<p name="Somatic.generateCnvPanelOfNormals.annotateIntervals.segmentalDuplicationTrack">
        <b>Somatic.generateCnvPanelOfNormals.annotateIntervals.segmentalDuplicationTrack</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to gatk AnnotateIntervals' `--segmenta-duplicarion-track` option.
</p>
<p name="Somatic.generateCnvPanelOfNormals.PONname">
        <b>Somatic.generateCnvPanelOfNormals.PONname</b><br />
        <i>String &mdash; Default: "PON"</i><br />
        The name the PON file should be given.
</p>
<p name="Somatic.outputDir">
        <b>Somatic.outputDir</b><br />
        <i>String &mdash; Default: "."</i><br />
        The directory the output should be written to.
</p>
<p name="Somatic.performCnvCalling">
        <b>Somatic.performCnvCalling</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not CNV calling should be performed.
</p>
<p name="Somatic.preprocessedIntervals">
        <b>Somatic.preprocessedIntervals</b><br />
        <i>File? &mdash; Default: None</i><br />
        The preprocessed intervals to be used for CNV calling. If not provided (and performCnvCalling is set to true) then this will be generated on the fly.
</p>
<p name="Somatic.regions">
        <b>Somatic.regions</b><br />
        <i>File? &mdash; Default: None</i><br />
        A bed file describing the regions to call variants for.
</p>
<p name="Somatic.runManta">
        <b>Somatic.runManta</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Whether or not manta should be run as part of the Strelka pipeline.
</p>
<p name="Somatic.runMutect2">
        <b>Somatic.runMutect2</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Whether or not to run Mutect2.
</p>
<p name="Somatic.runStrelka">
        <b>Somatic.runStrelka</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Whether or not to run Strelka.
</p>
<p name="Somatic.runVardict">
        <b>Somatic.runVardict</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Whether or not to run VarDict.
</p>
<p name="Somatic.sampleWorkflow.bqsr.splitSplicedReads">
        <b>Somatic.sampleWorkflow.bqsr.splitSplicedReads</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not gatk's SplitNCgarReads should be run to split spliced reads. This should be enabled for RNAseq samples.
</p>
<p name="Somatic.sampleWorkflow.bwaMem.sixtyFour">
        <b>Somatic.sampleWorkflow.bwaMem.sixtyFour</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not the index uses the '.64' suffixes.
</p>
<p name="Somatic.sampleWorkflow.bwaMem2Index">
        <b>Somatic.sampleWorkflow.bwaMem2Index</b><br />
        <i>struct(fastaFile : File, indexFiles : Array[File])? &mdash; Default: None</i><br />
        The bwa-mem2 index files. These or the bwaIndex should be provided.
</p>
<p name="Somatic.sampleWorkflow.bwamem2Mem.sixtyFour">
        <b>Somatic.sampleWorkflow.bwamem2Mem.sixtyFour</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not the index uses the '.64' suffixes.
</p>
<p name="Somatic.sampleWorkflow.metrics.ampliconIntervals">
        <b>Somatic.sampleWorkflow.metrics.ampliconIntervals</b><br />
        <i>File? &mdash; Default: None</i><br />
        An interval list describinig the coordinates of the amplicons sequenced. This should only be used for targeted sequencing or WES. Required if `ampliconIntervals` is defined.
</p>
<p name="Somatic.sampleWorkflow.metrics.refRefflat">
        <b>Somatic.sampleWorkflow.metrics.refRefflat</b><br />
        <i>File? &mdash; Default: None</i><br />
        A refflat file containing gene annotations. If defined RNA sequencing metrics will be collected.
</p>
<p name="Somatic.sampleWorkflow.metrics.strandedness">
        <b>Somatic.sampleWorkflow.metrics.strandedness</b><br />
        <i>String &mdash; Default: "None"</i><br />
        The strandedness of the RNA sequencing library preparation. One of "None" (unstranded), "FR" (forward-reverse: first read equal transcript) or "RF" (reverse-forward: second read equals transcript).
</p>
<p name="Somatic.sampleWorkflow.metrics.targetIntervals">
        <b>Somatic.sampleWorkflow.metrics.targetIntervals</b><br />
        <i>Array[File]+? &mdash; Default: None</i><br />
        An interval list describing the coordinates of the targets sequenced. This should only be used for targeted sequencing or WES. If defined targeted PCR metrics will be collected. Requires `ampliconIntervals` to also be defined.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.contaminations">
        <b>Somatic.sampleWorkflow.qualityControl.contaminations</b><br />
        <i>Array[String]+? &mdash; Default: None</i><br />
        Contaminants/adapters to be removed from the reads.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.readgroupName">
        <b>Somatic.sampleWorkflow.qualityControl.readgroupName</b><br />
        <i>String &mdash; Default: sub(basename(read1),"(\.fq)?(\.fastq)?(\.gz)?","")</i><br />
        The name of the readgroup.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.exclusionRegion">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.exclusionRegion</b><br />
        <i>File? &mdash; Default: None</i><br />
        A bed file describing regions to exclude.
</p>
<p name="Somatic.somaticVariantcalling.pairedTraining.exclusionRegion">
        <b>Somatic.somaticVariantcalling.pairedTraining.exclusionRegion</b><br />
        <i>File? &mdash; Default: None</i><br />
        A bed file describing regions to exclude.
</p>
<p name="Somatic.somaticVariantcalling.singleSomaticSeq.exclusionRegion">
        <b>Somatic.somaticVariantcalling.singleSomaticSeq.exclusionRegion</b><br />
        <i>File? &mdash; Default: None</i><br />
        A bed file describing regions to exclude.
</p>
<p name="Somatic.somaticVariantcalling.singleTraining.exclusionRegion">
        <b>Somatic.somaticVariantcalling.singleTraining.exclusionRegion</b><br />
        <i>File? &mdash; Default: None</i><br />
        A bed file describing regions to exclude.
</p>
<p name="Somatic.somaticVariantcalling.strelka.exome">
        <b>Somatic.somaticVariantcalling.strelka.exome</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not the data is from exome sequencing.
</p>
<p name="Somatic.somaticVariantcalling.strelka.rna">
        <b>Somatic.somaticVariantcalling.strelka.rna</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not the data is from RNA sequencing.
</p>

### Advanced inputs
<details>
<summary> Show/Hide </summary>
<p name="Somatic.bwaThreads">
        <b>Somatic.bwaThreads</b><br />
        <i>Int &mdash; Default: 4</i><br />
        The amount of threads for the alignment process
</p>
<p name="Somatic.cnvMinimumContigLength">
        <b>Somatic.cnvMinimumContigLength</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The minimum length for a contig to be included in the CNV plots.
</p>
<p name="Somatic.CNVs.caseSample.callCopyRatioSegments.javaXmx">
        <b>Somatic.CNVs.caseSample.callCopyRatioSegments.javaXmx</b><br />
        <i>String &mdash; Default: "2G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.caseSample.callCopyRatioSegments.memory">
        <b>Somatic.CNVs.caseSample.callCopyRatioSegments.memory</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.caseSample.callCopyRatioSegments.timeMinutes">
        <b>Somatic.CNVs.caseSample.callCopyRatioSegments.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.caseSample.collectAllelicCounts.javaXmx">
        <b>Somatic.CNVs.caseSample.collectAllelicCounts.javaXmx</b><br />
        <i>String &mdash; Default: "10G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.caseSample.collectAllelicCounts.memory">
        <b>Somatic.CNVs.caseSample.collectAllelicCounts.memory</b><br />
        <i>String &mdash; Default: "11G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.caseSample.collectAllelicCounts.timeMinutes">
        <b>Somatic.CNVs.caseSample.collectAllelicCounts.timeMinutes</b><br />
        <i>Int &mdash; Default: 120</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.caseSample.collectReadCounts.intervalMergingRule">
        <b>Somatic.CNVs.caseSample.collectReadCounts.intervalMergingRule</b><br />
        <i>String &mdash; Default: "OVERLAPPING_ONLY"</i><br />
        Equivalent to gatk CollectReadCounts' `--interval-merging-rule` option.
</p>
<p name="Somatic.CNVs.caseSample.collectReadCounts.javaXmx">
        <b>Somatic.CNVs.caseSample.collectReadCounts.javaXmx</b><br />
        <i>String &mdash; Default: "7G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.caseSample.collectReadCounts.memory">
        <b>Somatic.CNVs.caseSample.collectReadCounts.memory</b><br />
        <i>String &mdash; Default: "8G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.caseSample.collectReadCounts.timeMinutes">
        <b>Somatic.CNVs.caseSample.collectReadCounts.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inputBam,"G") * 5))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.caseSample.denoiseReadCounts.javaXmx">
        <b>Somatic.CNVs.caseSample.denoiseReadCounts.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.caseSample.denoiseReadCounts.memory">
        <b>Somatic.CNVs.caseSample.denoiseReadCounts.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.caseSample.denoiseReadCounts.timeMinutes">
        <b>Somatic.CNVs.caseSample.denoiseReadCounts.timeMinutes</b><br />
        <i>Int &mdash; Default: 5</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.caseSample.modelSegments.javaXmx">
        <b>Somatic.CNVs.caseSample.modelSegments.javaXmx</b><br />
        <i>String &mdash; Default: "10G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.caseSample.modelSegments.maximumNumberOfSmoothingIterations">
        <b>Somatic.CNVs.caseSample.modelSegments.maximumNumberOfSmoothingIterations</b><br />
        <i>Int &mdash; Default: 10</i><br />
        Equivalent to gatk ModelSeqments' `--maximum-number-of-smoothing-iterations` option.
</p>
<p name="Somatic.CNVs.caseSample.modelSegments.memory">
        <b>Somatic.CNVs.caseSample.modelSegments.memory</b><br />
        <i>String &mdash; Default: "11G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.caseSample.modelSegments.minimumTotalAlleleCountCase">
        <b>Somatic.CNVs.caseSample.modelSegments.minimumTotalAlleleCountCase</b><br />
        <i>Int &mdash; Default: if defined(normalAllelicCounts) then 0 else 30</i><br />
        Equivalent to gatk ModelSeqments' `--minimum-total-allele-count-case` option.
</p>
<p name="Somatic.CNVs.caseSample.modelSegments.timeMinutes">
        <b>Somatic.CNVs.caseSample.modelSegments.timeMinutes</b><br />
        <i>Int &mdash; Default: 60</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.caseSample.plotDenoisedCopyRatios.javaXmx">
        <b>Somatic.CNVs.caseSample.plotDenoisedCopyRatios.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.caseSample.plotDenoisedCopyRatios.memory">
        <b>Somatic.CNVs.caseSample.plotDenoisedCopyRatios.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.caseSample.plotDenoisedCopyRatios.timeMinutes">
        <b>Somatic.CNVs.caseSample.plotDenoisedCopyRatios.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.caseSample.plotModeledSegments.javaXmx">
        <b>Somatic.CNVs.caseSample.plotModeledSegments.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.caseSample.plotModeledSegments.memory">
        <b>Somatic.CNVs.caseSample.plotModeledSegments.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.caseSample.plotModeledSegments.timeMinutes">
        <b>Somatic.CNVs.caseSample.plotModeledSegments.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.controlSample.callCopyRatioSegments.javaXmx">
        <b>Somatic.CNVs.controlSample.callCopyRatioSegments.javaXmx</b><br />
        <i>String &mdash; Default: "2G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.controlSample.callCopyRatioSegments.memory">
        <b>Somatic.CNVs.controlSample.callCopyRatioSegments.memory</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.controlSample.callCopyRatioSegments.timeMinutes">
        <b>Somatic.CNVs.controlSample.callCopyRatioSegments.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.controlSample.collectAllelicCounts.javaXmx">
        <b>Somatic.CNVs.controlSample.collectAllelicCounts.javaXmx</b><br />
        <i>String &mdash; Default: "10G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.controlSample.collectAllelicCounts.memory">
        <b>Somatic.CNVs.controlSample.collectAllelicCounts.memory</b><br />
        <i>String &mdash; Default: "11G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.controlSample.collectAllelicCounts.timeMinutes">
        <b>Somatic.CNVs.controlSample.collectAllelicCounts.timeMinutes</b><br />
        <i>Int &mdash; Default: 120</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.controlSample.collectReadCounts.intervalMergingRule">
        <b>Somatic.CNVs.controlSample.collectReadCounts.intervalMergingRule</b><br />
        <i>String &mdash; Default: "OVERLAPPING_ONLY"</i><br />
        Equivalent to gatk CollectReadCounts' `--interval-merging-rule` option.
</p>
<p name="Somatic.CNVs.controlSample.collectReadCounts.javaXmx">
        <b>Somatic.CNVs.controlSample.collectReadCounts.javaXmx</b><br />
        <i>String &mdash; Default: "7G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.controlSample.collectReadCounts.memory">
        <b>Somatic.CNVs.controlSample.collectReadCounts.memory</b><br />
        <i>String &mdash; Default: "8G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.controlSample.collectReadCounts.timeMinutes">
        <b>Somatic.CNVs.controlSample.collectReadCounts.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inputBam,"G") * 5))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.controlSample.denoiseReadCounts.javaXmx">
        <b>Somatic.CNVs.controlSample.denoiseReadCounts.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.controlSample.denoiseReadCounts.memory">
        <b>Somatic.CNVs.controlSample.denoiseReadCounts.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.controlSample.denoiseReadCounts.timeMinutes">
        <b>Somatic.CNVs.controlSample.denoiseReadCounts.timeMinutes</b><br />
        <i>Int &mdash; Default: 5</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.controlSample.modelSegments.javaXmx">
        <b>Somatic.CNVs.controlSample.modelSegments.javaXmx</b><br />
        <i>String &mdash; Default: "10G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.controlSample.modelSegments.maximumNumberOfSmoothingIterations">
        <b>Somatic.CNVs.controlSample.modelSegments.maximumNumberOfSmoothingIterations</b><br />
        <i>Int &mdash; Default: 10</i><br />
        Equivalent to gatk ModelSeqments' `--maximum-number-of-smoothing-iterations` option.
</p>
<p name="Somatic.CNVs.controlSample.modelSegments.memory">
        <b>Somatic.CNVs.controlSample.modelSegments.memory</b><br />
        <i>String &mdash; Default: "11G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.controlSample.modelSegments.minimumTotalAlleleCountCase">
        <b>Somatic.CNVs.controlSample.modelSegments.minimumTotalAlleleCountCase</b><br />
        <i>Int &mdash; Default: if defined(normalAllelicCounts) then 0 else 30</i><br />
        Equivalent to gatk ModelSeqments' `--minimum-total-allele-count-case` option.
</p>
<p name="Somatic.CNVs.controlSample.modelSegments.timeMinutes">
        <b>Somatic.CNVs.controlSample.modelSegments.timeMinutes</b><br />
        <i>Int &mdash; Default: 60</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.controlSample.plotDenoisedCopyRatios.javaXmx">
        <b>Somatic.CNVs.controlSample.plotDenoisedCopyRatios.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.controlSample.plotDenoisedCopyRatios.memory">
        <b>Somatic.CNVs.controlSample.plotDenoisedCopyRatios.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.controlSample.plotDenoisedCopyRatios.timeMinutes">
        <b>Somatic.CNVs.controlSample.plotDenoisedCopyRatios.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.CNVs.controlSample.plotModeledSegments.javaXmx">
        <b>Somatic.CNVs.controlSample.plotModeledSegments.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.CNVs.controlSample.plotModeledSegments.memory">
        <b>Somatic.CNVs.controlSample.plotModeledSegments.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.CNVs.controlSample.plotModeledSegments.timeMinutes">
        <b>Somatic.CNVs.controlSample.plotModeledSegments.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.controlPostition.dockerImage">
        <b>Somatic.controlPostition.dockerImage</b><br />
        <i>String &mdash; Default: "python@sha256:e0f6a4df17d5707637fa3557ab266f44dddc46ebfc82b0f1dbe725103961da4e"</i><br />
        The docker image used for this task. Changing this may result in errors which the developers may choose not to address.
</p>
<p name="Somatic.convertDockerImagesFile.dockerImage">
        <b>Somatic.convertDockerImagesFile.dockerImage</b><br />
        <i>String &mdash; Default: "quay.io/biocontainers/biowdl-input-converter:0.2.1--py_0"</i><br />
        The docker image used for this task. Changing this may result in errors which the developers may choose not to address.
</p>
<p name="Somatic.convertDockerImagesFile.memory">
        <b>Somatic.convertDockerImagesFile.memory</b><br />
        <i>String &mdash; Default: "128M"</i><br />
        The maximum amount of memory the job will need.
</p>
<p name="Somatic.convertDockerImagesFile.outputJson">
        <b>Somatic.convertDockerImagesFile.outputJson</b><br />
        <i>String &mdash; Default: basename(yaml,"\.ya?ml$") + ".json"</i><br />
        The location the output JSON file should be written to.
</p>
<p name="Somatic.convertDockerImagesFile.timeMinutes">
        <b>Somatic.convertDockerImagesFile.timeMinutes</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.convertSampleConfig.checkFileMd5sums">
        <b>Somatic.convertSampleConfig.checkFileMd5sums</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not the MD5 sums of the files mentioned in the samplesheet should be checked.
</p>
<p name="Somatic.convertSampleConfig.memory">
        <b>Somatic.convertSampleConfig.memory</b><br />
        <i>String &mdash; Default: "128M"</i><br />
        The amount of memory needed for the job.
</p>
<p name="Somatic.convertSampleConfig.old">
        <b>Somatic.convertSampleConfig.old</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not the old samplesheet format should be used.
</p>
<p name="Somatic.convertSampleConfig.outputFile">
        <b>Somatic.convertSampleConfig.outputFile</b><br />
        <i>String &mdash; Default: "samplesheet.json"</i><br />
        The location the JSON representation of the samplesheet should be written to.
</p>
<p name="Somatic.convertSampleConfig.skipFileCheck">
        <b>Somatic.convertSampleConfig.skipFileCheck</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Whether or not the existance of the files mentioned in the samplesheet should be checked.
</p>
<p name="Somatic.convertSampleConfig.timeMinutes">
        <b>Somatic.convertSampleConfig.timeMinutes</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.generateCnvPanelOfNormals.annotateIntervals.featureQueryLookahead">
        <b>Somatic.generateCnvPanelOfNormals.annotateIntervals.featureQueryLookahead</b><br />
        <i>Int &mdash; Default: 1000000</i><br />
        Equivalent to gatk AnnotateIntervals' `--feature-query-lookahead` option.
</p>
<p name="Somatic.generateCnvPanelOfNormals.annotateIntervals.intervalMergingRule">
        <b>Somatic.generateCnvPanelOfNormals.annotateIntervals.intervalMergingRule</b><br />
        <i>String &mdash; Default: "OVERLAPPING_ONLY"</i><br />
        Equivalent to gatk AnnotateIntervals' `--interval-merging-rule` option.
</p>
<p name="Somatic.generateCnvPanelOfNormals.annotateIntervals.javaXmx">
        <b>Somatic.generateCnvPanelOfNormals.annotateIntervals.javaXmx</b><br />
        <i>String &mdash; Default: "2G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.generateCnvPanelOfNormals.annotateIntervals.memory">
        <b>Somatic.generateCnvPanelOfNormals.annotateIntervals.memory</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.generateCnvPanelOfNormals.annotateIntervals.timeMinutes">
        <b>Somatic.generateCnvPanelOfNormals.annotateIntervals.timeMinutes</b><br />
        <i>Int &mdash; Default: 5</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.generateCnvPanelOfNormals.collectReadCounts.intervalMergingRule">
        <b>Somatic.generateCnvPanelOfNormals.collectReadCounts.intervalMergingRule</b><br />
        <i>String &mdash; Default: "OVERLAPPING_ONLY"</i><br />
        Equivalent to gatk CollectReadCounts' `--interval-merging-rule` option.
</p>
<p name="Somatic.generateCnvPanelOfNormals.collectReadCounts.javaXmx">
        <b>Somatic.generateCnvPanelOfNormals.collectReadCounts.javaXmx</b><br />
        <i>String &mdash; Default: "7G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.generateCnvPanelOfNormals.collectReadCounts.memory">
        <b>Somatic.generateCnvPanelOfNormals.collectReadCounts.memory</b><br />
        <i>String &mdash; Default: "8G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.generateCnvPanelOfNormals.collectReadCounts.timeMinutes">
        <b>Somatic.generateCnvPanelOfNormals.collectReadCounts.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inputBam,"G") * 5))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.generateCnvPanelOfNormals.createReadCountPanelOfNormals.javaXmx">
        <b>Somatic.generateCnvPanelOfNormals.createReadCountPanelOfNormals.javaXmx</b><br />
        <i>String &mdash; Default: "7G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.generateCnvPanelOfNormals.createReadCountPanelOfNormals.memory">
        <b>Somatic.generateCnvPanelOfNormals.createReadCountPanelOfNormals.memory</b><br />
        <i>String &mdash; Default: "8G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.generateCnvPanelOfNormals.createReadCountPanelOfNormals.timeMinutes">
        <b>Somatic.generateCnvPanelOfNormals.createReadCountPanelOfNormals.timeMinutes</b><br />
        <i>Int &mdash; Default: 5</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.generateCnvPanelOfNormals.performExplicitGcCorrection">
        <b>Somatic.generateCnvPanelOfNormals.performExplicitGcCorrection</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Whether or not explicit GC correction should be used for PON generation. Setting this to false will also disable the creation of annotated intervals.
</p>
<p name="Somatic.generateCnvPanelOfNormals.preprocessIntervals.binLength">
        <b>Somatic.generateCnvPanelOfNormals.preprocessIntervals.binLength</b><br />
        <i>Int &mdash; Default: if defined(intervals) then 0 else 1000</i><br />
        The size of the bins to be created. Should be 0 for targeted/exome sequencing.
</p>
<p name="Somatic.generateCnvPanelOfNormals.preprocessIntervals.intervalMergingRule">
        <b>Somatic.generateCnvPanelOfNormals.preprocessIntervals.intervalMergingRule</b><br />
        <i>String &mdash; Default: "OVERLAPPING_ONLY"</i><br />
        Equivalent to gatk PreprocessIntervals' `--interval-merging-rule` option.
</p>
<p name="Somatic.generateCnvPanelOfNormals.preprocessIntervals.javaXmx">
        <b>Somatic.generateCnvPanelOfNormals.preprocessIntervals.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.generateCnvPanelOfNormals.preprocessIntervals.memory">
        <b>Somatic.generateCnvPanelOfNormals.preprocessIntervals.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.generateCnvPanelOfNormals.preprocessIntervals.padding">
        <b>Somatic.generateCnvPanelOfNormals.preprocessIntervals.padding</b><br />
        <i>Int &mdash; Default: if defined(intervals) then 250 else 0</i><br />
        The padding to be added to the bins. Should be 0 if contiguos binning is used, eg with WGS.
</p>
<p name="Somatic.generateCnvPanelOfNormals.preprocessIntervals.timeMinutes">
        <b>Somatic.generateCnvPanelOfNormals.preprocessIntervals.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(referenceFasta,"G") * 6))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.multiqcTask.clConfig">
        <b>Somatic.multiqcTask.clConfig</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--cl-config` option.
</p>
<p name="Somatic.multiqcTask.comment">
        <b>Somatic.multiqcTask.comment</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--comment` option.
</p>
<p name="Somatic.multiqcTask.config">
        <b>Somatic.multiqcTask.config</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--config` option.
</p>
<p name="Somatic.multiqcTask.dataDir">
        <b>Somatic.multiqcTask.dataDir</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether to output a data dir. Sets `--data-dir` or `--no-data-dir` flag.
</p>
<p name="Somatic.multiqcTask.dataFormat">
        <b>Somatic.multiqcTask.dataFormat</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--data-format` option.
</p>
<p name="Somatic.multiqcTask.dirs">
        <b>Somatic.multiqcTask.dirs</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to MultiQC's `--dirs` flag.
</p>
<p name="Somatic.multiqcTask.dirsDepth">
        <b>Somatic.multiqcTask.dirsDepth</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--dirs-depth` option.
</p>
<p name="Somatic.multiqcTask.exclude">
        <b>Somatic.multiqcTask.exclude</b><br />
        <i>Array[String]+? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--exclude` option.
</p>
<p name="Somatic.multiqcTask.export">
        <b>Somatic.multiqcTask.export</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to MultiQC's `--export` flag.
</p>
<p name="Somatic.multiqcTask.fileList">
        <b>Somatic.multiqcTask.fileList</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--file-list` option.
</p>
<p name="Somatic.multiqcTask.fileName">
        <b>Somatic.multiqcTask.fileName</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--filename` option.
</p>
<p name="Somatic.multiqcTask.flat">
        <b>Somatic.multiqcTask.flat</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to MultiQC's `--flat` flag.
</p>
<p name="Somatic.multiqcTask.force">
        <b>Somatic.multiqcTask.force</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to MultiQC's `--force` flag.
</p>
<p name="Somatic.multiqcTask.fullNames">
        <b>Somatic.multiqcTask.fullNames</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to MultiQC's `--fullnames` flag.
</p>
<p name="Somatic.multiqcTask.ignore">
        <b>Somatic.multiqcTask.ignore</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--ignore` option.
</p>
<p name="Somatic.multiqcTask.ignoreSamples">
        <b>Somatic.multiqcTask.ignoreSamples</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--ignore-samples` option.
</p>
<p name="Somatic.multiqcTask.interactive">
        <b>Somatic.multiqcTask.interactive</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to MultiQC's `--interactive` flag.
</p>
<p name="Somatic.multiqcTask.lint">
        <b>Somatic.multiqcTask.lint</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to MultiQC's `--lint` flag.
</p>
<p name="Somatic.multiqcTask.megaQCUpload">
        <b>Somatic.multiqcTask.megaQCUpload</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Opposite to MultiQC's `--no-megaqc-upload` flag.
</p>
<p name="Somatic.multiqcTask.memory">
        <b>Somatic.multiqcTask.memory</b><br />
        <i>String? &mdash; Default: None</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.multiqcTask.module">
        <b>Somatic.multiqcTask.module</b><br />
        <i>Array[String]+? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--module` option.
</p>
<p name="Somatic.multiqcTask.pdf">
        <b>Somatic.multiqcTask.pdf</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to MultiQC's `--pdf` flag.
</p>
<p name="Somatic.multiqcTask.sampleNames">
        <b>Somatic.multiqcTask.sampleNames</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--sample-names` option.
</p>
<p name="Somatic.multiqcTask.tag">
        <b>Somatic.multiqcTask.tag</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--tag` option.
</p>
<p name="Somatic.multiqcTask.template">
        <b>Somatic.multiqcTask.template</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--template` option.
</p>
<p name="Somatic.multiqcTask.timeMinutes">
        <b>Somatic.multiqcTask.timeMinutes</b><br />
        <i>Int &mdash; Default: 2 + ceil((size(reports,"G") * 8))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.multiqcTask.title">
        <b>Somatic.multiqcTask.title</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to MultiQC's `--title` option.
</p>
<p name="Somatic.multiqcTask.zipDataDir">
        <b>Somatic.multiqcTask.zipDataDir</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to MultiQC's `--zip-data-dir` flag.
</p>
<p name="Somatic.platform">
        <b>Somatic.platform</b><br />
        <i>String &mdash; Default: "illumina"</i><br />
        The platform used for sequencing.
</p>
<p name="Somatic.runCombineVariants">
        <b>Somatic.runCombineVariants</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not to combine the variant calling results into one VCF file.
</p>
<p name="Somatic.sampleWorkflow.bqsr.applyBqsr.javaXmxMb">
        <b>Somatic.sampleWorkflow.bqsr.applyBqsr.javaXmxMb</b><br />
        <i>Int &mdash; Default: 2048</i><br />
        The maximum memory available to the program in megabytes. Should be lower than `memoryMb` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.bqsr.applyBqsr.memoryMb">
        <b>Somatic.sampleWorkflow.bqsr.applyBqsr.memoryMb</b><br />
        <i>Int &mdash; Default: javaXmxMb + 512</i><br />
        The amount of memory this job will use in megabytes.
</p>
<p name="Somatic.sampleWorkflow.bqsr.baseRecalibrator.javaXmxMb">
        <b>Somatic.sampleWorkflow.bqsr.baseRecalibrator.javaXmxMb</b><br />
        <i>Int &mdash; Default: 1024</i><br />
        The maximum memory available to the program in megabytes. Should be lower than `memoryMb` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.bqsr.baseRecalibrator.knownIndelsSitesVCFIndexes">
        <b>Somatic.sampleWorkflow.bqsr.baseRecalibrator.knownIndelsSitesVCFIndexes</b><br />
        <i>Array[File] &mdash; Default: []</i><br />
        The indexed for the known variant VCFs.
</p>
<p name="Somatic.sampleWorkflow.bqsr.baseRecalibrator.knownIndelsSitesVCFs">
        <b>Somatic.sampleWorkflow.bqsr.baseRecalibrator.knownIndelsSitesVCFs</b><br />
        <i>Array[File] &mdash; Default: []</i><br />
        VCF files with known indels.
</p>
<p name="Somatic.sampleWorkflow.bqsr.baseRecalibrator.memoryMb">
        <b>Somatic.sampleWorkflow.bqsr.baseRecalibrator.memoryMb</b><br />
        <i>Int &mdash; Default: javaXmxMb + 512</i><br />
        The amount of memory this job will use in megabytes.
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBamFiles.compressionLevel">
        <b>Somatic.sampleWorkflow.bqsr.gatherBamFiles.compressionLevel</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The compression level of the output BAM.
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBamFiles.createMd5File">
        <b>Somatic.sampleWorkflow.bqsr.gatherBamFiles.createMd5File</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        ???
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBamFiles.javaXmxMb">
        <b>Somatic.sampleWorkflow.bqsr.gatherBamFiles.javaXmxMb</b><br />
        <i>Int &mdash; Default: 1024</i><br />
        The maximum memory available to the program in megabytes. Should be lower than `memoryMb` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBamFiles.memoryMb">
        <b>Somatic.sampleWorkflow.bqsr.gatherBamFiles.memoryMb</b><br />
        <i>Int &mdash; Default: javaXmxMb + 512</i><br />
        The amount of memory this job will use in megabytes.
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBamFiles.timeMinutes">
        <b>Somatic.sampleWorkflow.bqsr.gatherBamFiles.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inputBams,"G") * 1))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBqsr.javaXmxMb">
        <b>Somatic.sampleWorkflow.bqsr.gatherBqsr.javaXmxMb</b><br />
        <i>Int &mdash; Default: 256</i><br />
        The maximum memory available to the program in megabytes. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBqsr.memoryMb">
        <b>Somatic.sampleWorkflow.bqsr.gatherBqsr.memoryMb</b><br />
        <i>Int &mdash; Default: 256 + javaXmxMb</i><br />
        The amount of memory this job will use in megabytes.
</p>
<p name="Somatic.sampleWorkflow.bqsr.gatherBqsr.timeMinutes">
        <b>Somatic.sampleWorkflow.bqsr.gatherBqsr.timeMinutes</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.bqsr.splitNCigarReads.javaXmx">
        <b>Somatic.sampleWorkflow.bqsr.splitNCigarReads.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.bqsr.splitNCigarReads.memory">
        <b>Somatic.sampleWorkflow.bqsr.splitNCigarReads.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.bwaMem.compressionLevel">
        <b>Somatic.sampleWorkflow.bwaMem.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The compression level of the output BAM.
</p>
<p name="Somatic.sampleWorkflow.bwaMem.memoryGb">
        <b>Somatic.sampleWorkflow.bwaMem.memoryGb</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The amount of memory this job will use in gigabytes.
</p>
<p name="Somatic.sampleWorkflow.bwaMem.sortMemoryPerThreadGb">
        <b>Somatic.sampleWorkflow.bwaMem.sortMemoryPerThreadGb</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The amount of memory for each sorting thread in gigabytes.
</p>
<p name="Somatic.sampleWorkflow.bwaMem.sortThreads">
        <b>Somatic.sampleWorkflow.bwaMem.sortThreads</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The number of threads to use for sorting.
</p>
<p name="Somatic.sampleWorkflow.bwaMem.timeMinutes">
        <b>Somatic.sampleWorkflow.bwaMem.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size([read1, read2],"G") * 220 / threads))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.bwamem2Mem.compressionLevel">
        <b>Somatic.sampleWorkflow.bwamem2Mem.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The compression level of the output BAM.
</p>
<p name="Somatic.sampleWorkflow.bwamem2Mem.memoryGb">
        <b>Somatic.sampleWorkflow.bwamem2Mem.memoryGb</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The amount of memory this job will use in gigabytes.
</p>
<p name="Somatic.sampleWorkflow.bwamem2Mem.sortMemoryPerThreadGb">
        <b>Somatic.sampleWorkflow.bwamem2Mem.sortMemoryPerThreadGb</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The amount of memory for each sorting thread in gigabytes.
</p>
<p name="Somatic.sampleWorkflow.bwamem2Mem.sortThreads">
        <b>Somatic.sampleWorkflow.bwamem2Mem.sortThreads</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The number of threads to use for sorting.
</p>
<p name="Somatic.sampleWorkflow.bwamem2Mem.timeMinutes">
        <b>Somatic.sampleWorkflow.bwamem2Mem.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size([read1, read2],"G") * 220 / threads))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.markdup.compressionLevel">
        <b>Somatic.sampleWorkflow.markdup.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        Compression level from 0 (uncompressed) to 9 (best).
</p>
<p name="Somatic.sampleWorkflow.markdup.hashTableSize">
        <b>Somatic.sampleWorkflow.markdup.hashTableSize</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Sets sambamba's hash table size.
</p>
<p name="Somatic.sampleWorkflow.markdup.ioBufferSize">
        <b>Somatic.sampleWorkflow.markdup.ioBufferSize</b><br />
        <i>Int &mdash; Default: 128</i><br />
        The amount of mb allocated to each IO buffer. Sambamba uses two IO buffers.
</p>
<p name="Somatic.sampleWorkflow.markdup.memoryMb">
        <b>Somatic.sampleWorkflow.markdup.memoryMb</b><br />
        <i>Int &mdash; Default: 1024 + sortBufferSize + 2 * ioBufferSize</i><br />
        The amount of memory available to the job in megabytes.
</p>
<p name="Somatic.sampleWorkflow.markdup.overFlowListSize">
        <b>Somatic.sampleWorkflow.markdup.overFlowListSize</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Sets sambamba's overflow list size.
</p>
<p name="Somatic.sampleWorkflow.markdup.removeDuplicates">
        <b>Somatic.sampleWorkflow.markdup.removeDuplicates</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether to remove the duplicates (instead of only marking them).
</p>
<p name="Somatic.sampleWorkflow.markdup.sortBufferSize">
        <b>Somatic.sampleWorkflow.markdup.sortBufferSize</b><br />
        <i>Int &mdash; Default: 2048</i><br />
        The amount of mb allocated to the sort buffer.
</p>
<p name="Somatic.sampleWorkflow.markdup.threads">
        <b>Somatic.sampleWorkflow.markdup.threads</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The number of threads that will be used for this task.
</p>
<p name="Somatic.sampleWorkflow.markdup.timeMinutes">
        <b>Somatic.sampleWorkflow.markdup.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inputBams,"G") * 8)) / threads</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.metrics.ampliconIntervalsLists.javaXmx">
        <b>Somatic.sampleWorkflow.metrics.ampliconIntervalsLists.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.metrics.ampliconIntervalsLists.memory">
        <b>Somatic.sampleWorkflow.metrics.ampliconIntervalsLists.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.metrics.ampliconIntervalsLists.timeMinutes">
        <b>Somatic.sampleWorkflow.metrics.ampliconIntervalsLists.timeMinutes</b><br />
        <i>Int &mdash; Default: 5</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.metrics.collectAlignmentSummaryMetrics">
        <b>Somatic.sampleWorkflow.metrics.collectAlignmentSummaryMetrics</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=CollectAlignmentSummaryMetrics` argument in Picard.
</p>
<p name="Somatic.sampleWorkflow.metrics.Flagstat.memory">
        <b>Somatic.sampleWorkflow.metrics.Flagstat.memory</b><br />
        <i>String &mdash; Default: "256M"</i><br />
        The amount of memory needed for the job.
</p>
<p name="Somatic.sampleWorkflow.metrics.Flagstat.timeMinutes">
        <b>Somatic.sampleWorkflow.metrics.Flagstat.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputBam,"G"))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.metrics.meanQualityByCycle">
        <b>Somatic.sampleWorkflow.metrics.meanQualityByCycle</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=MeanQualityByCycle` argument in Picard.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.collectBaseDistributionByCycle">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.collectBaseDistributionByCycle</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=CollectBaseDistributionByCycle` argument.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.collectGcBiasMetrics">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.collectGcBiasMetrics</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=CollectGcBiasMetrics` argument.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.collectInsertSizeMetrics">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.collectInsertSizeMetrics</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=CollectInsertSizeMetrics` argument.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.collectQualityYieldMetrics">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.collectQualityYieldMetrics</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=CollectQualityYieldMetrics` argument.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.collectSequencingArtifactMetrics">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.collectSequencingArtifactMetrics</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=CollectSequencingArtifactMetrics` argument.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.javaXmxMb">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.javaXmxMb</b><br />
        <i>Int &mdash; Default: 3072</i><br />
        The maximum memory available to the program in megabytes. Should be lower than `memoryMb` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.memoryMb">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.memoryMb</b><br />
        <i>Int &mdash; Default: javaXmxMb + 512</i><br />
        The amount of memory this job will use in megabytes.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.qualityScoreDistribution">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.qualityScoreDistribution</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to the `PROGRAM=QualityScoreDistribution` argument.
</p>
<p name="Somatic.sampleWorkflow.metrics.picardMetrics.timeMinutes">
        <b>Somatic.sampleWorkflow.metrics.picardMetrics.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(referenceFasta,"G") * 3 * 2)) + ceil((size(inputBam,"G") * 6))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.metrics.rnaSeqMetrics.javaXmx">
        <b>Somatic.sampleWorkflow.metrics.rnaSeqMetrics.javaXmx</b><br />
        <i>String &mdash; Default: "8G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.metrics.rnaSeqMetrics.memory">
        <b>Somatic.sampleWorkflow.metrics.rnaSeqMetrics.memory</b><br />
        <i>String &mdash; Default: "9G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.metrics.rnaSeqMetrics.timeMinutes">
        <b>Somatic.sampleWorkflow.metrics.rnaSeqMetrics.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inputBam,"G") * 12))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.metrics.targetIntervalsLists.javaXmx">
        <b>Somatic.sampleWorkflow.metrics.targetIntervalsLists.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.metrics.targetIntervalsLists.memory">
        <b>Somatic.sampleWorkflow.metrics.targetIntervalsLists.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.metrics.targetIntervalsLists.timeMinutes">
        <b>Somatic.sampleWorkflow.metrics.targetIntervalsLists.timeMinutes</b><br />
        <i>Int &mdash; Default: 5</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.metrics.targetMetrics.javaXmx">
        <b>Somatic.sampleWorkflow.metrics.targetMetrics.javaXmx</b><br />
        <i>String &mdash; Default: "3G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.metrics.targetMetrics.memory">
        <b>Somatic.sampleWorkflow.metrics.targetMetrics.memory</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.metrics.targetMetrics.timeMinutes">
        <b>Somatic.sampleWorkflow.metrics.targetMetrics.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inputBam,"G") * 6))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.bwa">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.bwa</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --bwa flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.colorspace">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.colorspace</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --colorspace flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.compressionLevel">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The compression level if gzipped output is used.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.cores">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.cores</b><br />
        <i>Int &mdash; Default: 4</i><br />
        The number of cores to use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.cut">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.cut</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --cut option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.discardTrimmed">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.discardTrimmed</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --quality-cutoff option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.discardUntrimmed">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.discardUntrimmed</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --discard-untrimmed option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.doubleEncode">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.doubleEncode</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --double-encode flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.errorRate">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.errorRate</b><br />
        <i>Float? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --error-rate option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.front">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.front</b><br />
        <i>Array[String] &mdash; Default: []</i><br />
        A list of 5' ligated adapter sequences to be cut from the given first or single end fastq file.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.frontRead2">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.frontRead2</b><br />
        <i>Array[String] &mdash; Default: []</i><br />
        A list of 5' ligated adapter sequences to be cut from the given second end fastq file.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.infoFilePath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.infoFilePath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --info-file option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.interleaved">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.interleaved</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --interleaved flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.length">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.length</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --length option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.lengthTag">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.lengthTag</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --length-tag option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.maq">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.maq</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --maq flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.maskAdapter">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.maskAdapter</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --mask-adapter flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.matchReadWildcards">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.matchReadWildcards</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --match-read-wildcards flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.maximumLength">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.maximumLength</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --maximum-length option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.maxN">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.maxN</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --max-n option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.memory">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.memory</b><br />
        <i>String &mdash; Default: "~{300 + 100 * cores}M"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.minimumLength">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.minimumLength</b><br />
        <i>Int? &mdash; Default: 2</i><br />
        Equivalent to cutadapt's --minimum-length option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.nextseqTrim">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.nextseqTrim</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --nextseq-trim option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.noIndels">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.noIndels</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --no-indels flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.noMatchAdapterWildcards">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.noMatchAdapterWildcards</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --no-match-adapter-wildcards flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.noTrim">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.noTrim</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --no-trim flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.noZeroCap">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.noZeroCap</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --no-zero-cap flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.overlap">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.overlap</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --overlap option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.pairFilter">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.pairFilter</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --pair-filter option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.prefix">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.prefix</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --prefix option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.qualityBase">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.qualityBase</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --quality-base option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.qualityCutoff">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.qualityCutoff</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --quality-cutoff option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.restFilePath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.restFilePath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --rest-file option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.stripF3">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.stripF3</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --strip-f3 flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.stripSuffix">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.stripSuffix</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --strip-suffix option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.suffix">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.suffix</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --suffix option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.timeMinutes">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size([read1, read2],"G") * 12.0 / cores))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.times">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.times</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --times option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.tooLongOutputPath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.tooLongOutputPath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --too-long-output option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.tooLongPairedOutputPath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.tooLongPairedOutputPath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --too-long-paired-output option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.tooShortOutputPath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.tooShortOutputPath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --too-short-output option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.tooShortPairedOutputPath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.tooShortPairedOutputPath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --too-short-paired-output option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.trimN">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.trimN</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --trim-n flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.untrimmedOutputPath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.untrimmedOutputPath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --untrimmed-output option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.untrimmedPairedOutputPath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.untrimmedPairedOutputPath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --untrimmed-paired-output option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.wildcardFilePath">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.wildcardFilePath</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --wildcard-file option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.Cutadapt.zeroCap">
        <b>Somatic.sampleWorkflow.qualityControl.Cutadapt.zeroCap</b><br />
        <i>Boolean? &mdash; Default: None</i><br />
        Equivalent to cutadapt's --zero-cap flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.extractFastqcZip">
        <b>Somatic.sampleWorkflow.qualityControl.extractFastqcZip</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether to extract Fastqc's report zip files.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.adapters">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.adapters</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --adapters option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.casava">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.casava</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --casava flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.contaminants">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.contaminants</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --contaminants option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.dir">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.dir</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --dir option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.format">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.format</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --format option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.javaXmx">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.javaXmx</b><br />
        <i>String &mdash; Default: "1750M"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.kmers">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.kmers</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --kmers option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.limits">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.limits</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --limits option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.memory">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.memory</b><br />
        <i>String &mdash; Default: "2G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.minLength">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.minLength</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --min_length option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.nano">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.nano</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nano flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.noFilter">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.noFilter</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nofilter flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.nogroup">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.nogroup</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nogroup flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.threads">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of cores to use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1.timeMinutes">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(seqFile,"G")) * 4</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.adapters">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.adapters</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --adapters option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.casava">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.casava</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --casava flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.contaminants">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.contaminants</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --contaminants option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.dir">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.dir</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --dir option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.format">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.format</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --format option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.javaXmx">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.javaXmx</b><br />
        <i>String &mdash; Default: "1750M"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.kmers">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.kmers</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --kmers option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.limits">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.limits</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --limits option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.memory">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.memory</b><br />
        <i>String &mdash; Default: "2G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.minLength">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.minLength</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --min_length option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.nano">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.nano</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nano flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.noFilter">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.noFilter</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nofilter flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.nogroup">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.nogroup</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nogroup flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.threads">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of cores to use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead1After.timeMinutes">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead1After.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(seqFile,"G")) * 4</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.adapters">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.adapters</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --adapters option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.casava">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.casava</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --casava flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.contaminants">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.contaminants</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --contaminants option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.dir">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.dir</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --dir option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.format">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.format</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --format option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.javaXmx">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.javaXmx</b><br />
        <i>String &mdash; Default: "1750M"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.kmers">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.kmers</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --kmers option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.limits">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.limits</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --limits option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.memory">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.memory</b><br />
        <i>String &mdash; Default: "2G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.minLength">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.minLength</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --min_length option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.nano">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.nano</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nano flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.noFilter">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.noFilter</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nofilter flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.nogroup">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.nogroup</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nogroup flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.threads">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of cores to use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2.timeMinutes">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(seqFile,"G")) * 4</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.adapters">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.adapters</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --adapters option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.casava">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.casava</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --casava flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.contaminants">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.contaminants</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --contaminants option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.dir">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.dir</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --dir option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.format">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.format</b><br />
        <i>String? &mdash; Default: None</i><br />
        Equivalent to fastqc's --format option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.javaXmx">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.javaXmx</b><br />
        <i>String &mdash; Default: "1750M"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.kmers">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.kmers</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --kmers option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.limits">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.limits</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to fastqc's --limits option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.memory">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.memory</b><br />
        <i>String &mdash; Default: "2G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.minLength">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.minLength</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to fastqc's --min_length option.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.nano">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.nano</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nano flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.noFilter">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.noFilter</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nofilter flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.nogroup">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.nogroup</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to fastqc's --nogroup flag.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.threads">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of cores to use.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.FastqcRead2After.timeMinutes">
        <b>Somatic.sampleWorkflow.qualityControl.FastqcRead2After.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(seqFile,"G")) * 4</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.sampleWorkflow.qualityControl.runAdapterClipping">
        <b>Somatic.sampleWorkflow.qualityControl.runAdapterClipping</b><br />
        <i>Boolean &mdash; Default: defined(adapterForward) || defined(adapterReverse) || length(select_first([contaminations, []])) > 0</i><br />
        Whether or not adapters should be removed from the reads.
</p>
<p name="Somatic.scatterList.memory">
        <b>Somatic.scatterList.memory</b><br />
        <i>String &mdash; Default: "256M"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.scatterList.prefix">
        <b>Somatic.scatterList.prefix</b><br />
        <i>String &mdash; Default: "scatters/scatter-"</i><br />
        The prefix of the ouput files. Output will be named like: <PREFIX><N>.bed, in which N is an incrementing number. Default 'scatter-'.
</p>
<p name="Somatic.scatterList.splitContigs">
        <b>Somatic.scatterList.splitContigs</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        If set, contigs are allowed to be split up over multiple files.
</p>
<p name="Somatic.scatterList.timeMinutes">
        <b>Somatic.scatterList.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.scatterSize">
        <b>Somatic.scatterSize</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The size of the scattered regions in bases for the GATK subworkflows. Scattering is used to speed up certain processes. The genome will be seperated into multiple chunks (scatters) which will be processed in their own job, allowing for parallel processing. Higher values will result in a lower number of jobs. The optimal value here will depend on the available resources.
</p>
<p name="Somatic.scatterSizeMillions">
        <b>Somatic.scatterSizeMillions</b><br />
        <i>Int &mdash; Default: 1000</i><br />
        Same as scatterSize, but is multiplied by 1000000 to get scatterSize. This allows for setting larger values more easily.
</p>
<p name="Somatic.somaticVariantcalling.combineVariants.dockerImage">
        <b>Somatic.somaticVariantcalling.combineVariants.dockerImage</b><br />
        <i>String &mdash; Default: "broadinstitute/gatk3:3.8-1"</i><br />
        The docker image used for this task. Changing this may result in errors which the developers may choose not to address.
</p>
<p name="Somatic.somaticVariantcalling.combineVariants.filteredRecordsMergeType">
        <b>Somatic.somaticVariantcalling.combineVariants.filteredRecordsMergeType</b><br />
        <i>String &mdash; Default: "KEEP_IF_ANY_UNFILTERED"</i><br />
        Equivalent to CombineVariants' `--filteredrecordsmergetype` option.
</p>
<p name="Somatic.somaticVariantcalling.combineVariants.genotypeMergeOption">
        <b>Somatic.somaticVariantcalling.combineVariants.genotypeMergeOption</b><br />
        <i>String &mdash; Default: "UNIQUIFY"</i><br />
        Equivalent to CombineVariants' `--genotypemergeoption` option.
</p>
<p name="Somatic.somaticVariantcalling.combineVariants.javaXmx">
        <b>Somatic.somaticVariantcalling.combineVariants.javaXmx</b><br />
        <i>String &mdash; Default: "12G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.combineVariants.memory">
        <b>Somatic.somaticVariantcalling.combineVariants.memory</b><br />
        <i>String &mdash; Default: "13G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.combineVariants.timeMinutes">
        <b>Somatic.somaticVariantcalling.combineVariants.timeMinutes</b><br />
        <i>Int &mdash; Default: 180</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.indelIndex.timeMinutes">
        <b>Somatic.somaticVariantcalling.indelIndex.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputFile,"G"))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.calculateContamination.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.calculateContamination.javaXmx</b><br />
        <i>String &mdash; Default: "12G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.calculateContamination.memory">
        <b>Somatic.somaticVariantcalling.mutect2.calculateContamination.memory</b><br />
        <i>String &mdash; Default: "13G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.calculateContamination.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.calculateContamination.timeMinutes</b><br />
        <i>Int &mdash; Default: 180</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.filterMutectCalls.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.filterMutectCalls.javaXmx</b><br />
        <i>String &mdash; Default: "12G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.filterMutectCalls.memory">
        <b>Somatic.somaticVariantcalling.mutect2.filterMutectCalls.memory</b><br />
        <i>String &mdash; Default: "13G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.filterMutectCalls.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.filterMutectCalls.timeMinutes</b><br />
        <i>Int &mdash; Default: 60</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.filterMutectCalls.uniqueAltReadCount">
        <b>Somatic.somaticVariantcalling.mutect2.filterMutectCalls.uniqueAltReadCount</b><br />
        <i>Int &mdash; Default: 4</i><br />
        Equivalent to FilterMutectCalls' `--unique-alt-read-count` option.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.gatherVcfs.compressionLevel">
        <b>Somatic.somaticVariantcalling.mutect2.gatherVcfs.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The compression level at which the BAM files are written.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.gatherVcfs.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.gatherVcfs.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.gatherVcfs.memory">
        <b>Somatic.somaticVariantcalling.mutect2.gatherVcfs.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.gatherVcfs.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.gatherVcfs.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputVCFs,"G")) * 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.gatherVcfs.useJdkDeflater">
        <b>Somatic.somaticVariantcalling.mutect2.gatherVcfs.useJdkDeflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java deflator to compress the BAM files. False uses the optimized intel deflater.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.gatherVcfs.useJdkInflater">
        <b>Somatic.somaticVariantcalling.mutect2.gatherVcfs.useJdkInflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java inflater. False, uses the optimized intel inflater.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.getPileupSummariesNormal.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.getPileupSummariesNormal.javaXmx</b><br />
        <i>String &mdash; Default: "12G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.getPileupSummariesNormal.memory">
        <b>Somatic.somaticVariantcalling.mutect2.getPileupSummariesNormal.memory</b><br />
        <i>String &mdash; Default: "13G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.getPileupSummariesNormal.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.getPileupSummariesNormal.timeMinutes</b><br />
        <i>Int &mdash; Default: 120</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.getPileupSummariesTumor.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.getPileupSummariesTumor.javaXmx</b><br />
        <i>String &mdash; Default: "12G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.getPileupSummariesTumor.memory">
        <b>Somatic.somaticVariantcalling.mutect2.getPileupSummariesTumor.memory</b><br />
        <i>String &mdash; Default: "13G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.getPileupSummariesTumor.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.getPileupSummariesTumor.timeMinutes</b><br />
        <i>Int &mdash; Default: 120</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.learnReadOrientationModel.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.learnReadOrientationModel.javaXmx</b><br />
        <i>String &mdash; Default: "12G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.learnReadOrientationModel.memory">
        <b>Somatic.somaticVariantcalling.mutect2.learnReadOrientationModel.memory</b><br />
        <i>String &mdash; Default: "13G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.learnReadOrientationModel.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.learnReadOrientationModel.timeMinutes</b><br />
        <i>Int &mdash; Default: 120</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mergeStats.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.mergeStats.javaXmx</b><br />
        <i>String &mdash; Default: "14G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mergeStats.memory">
        <b>Somatic.somaticVariantcalling.mutect2.mergeStats.memory</b><br />
        <i>String &mdash; Default: "15G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mergeStats.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.mergeStats.timeMinutes</b><br />
        <i>Int &mdash; Default: 30</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.f1r2TarGz">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.f1r2TarGz</b><br />
        <i>String &mdash; Default: "f1r2.tar.gz"</i><br />
        Equivalent to Mutect2's `--f1r2-tar-gz` option.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.germlineResource">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.germlineResource</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to Mutect2's `--germline-resource` option.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.germlineResourceIndex">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.germlineResourceIndex</b><br />
        <i>File? &mdash; Default: None</i><br />
        The index for the germline resource.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.javaXmx">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.memory">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.outputStats">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.outputStats</b><br />
        <i>String &mdash; Default: outputVcf + ".stats"</i><br />
        The location the output statistics should be written to.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.panelOfNormals">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.panelOfNormals</b><br />
        <i>File? &mdash; Default: None</i><br />
        Equivalent to Mutect2's `--panel-of-normals` option.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.panelOfNormalsIndex">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.panelOfNormalsIndex</b><br />
        <i>File? &mdash; Default: None</i><br />
        The index for the panel of normals.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.mutect2.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.mutect2.timeMinutes</b><br />
        <i>Int &mdash; Default: 240</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.scatterList.memory">
        <b>Somatic.somaticVariantcalling.mutect2.scatterList.memory</b><br />
        <i>String &mdash; Default: "256M"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.scatterList.prefix">
        <b>Somatic.somaticVariantcalling.mutect2.scatterList.prefix</b><br />
        <i>String &mdash; Default: "scatters/scatter-"</i><br />
        The prefix of the ouput files. Output will be named like: <PREFIX><N>.bed, in which N is an incrementing number. Default 'scatter-'.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.scatterList.timeMinutes">
        <b>Somatic.somaticVariantcalling.mutect2.scatterList.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.scatterSize">
        <b>Somatic.somaticVariantcalling.mutect2.scatterSize</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The size of the scattered regions in bases. Scattering is used to speed up certain processes. The genome will be seperated into multiple chunks (scatters) which will be processed in their own job, allowing for parallel processing. Higher values will result in a lower number of jobs. The optimal value here will depend on the available resources.
</p>
<p name="Somatic.somaticVariantcalling.mutect2.scatterSizeMillions">
        <b>Somatic.somaticVariantcalling.mutect2.scatterSizeMillions</b><br />
        <i>Int &mdash; Default: 1000</i><br />
        Same as scatterSize, but is multiplied by 1000000 to get scatterSize. This allows for setting larger values more easily
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.jsmVCF">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.jsmVCF</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF as produced by jsm.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.lofreqIndel">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.lofreqIndel</b><br />
        <i>File? &mdash; Default: None</i><br />
        An indel VCF as produced by lofreq.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.lofreqSNV">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.lofreqSNV</b><br />
        <i>File? &mdash; Default: None</i><br />
        An SNV VCF as produced by lofreq.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.museVCF">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.museVCF</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF as produced by muse.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.scalpelVCF">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.scalpelVCF</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF as produced by scalpel.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.somaticsniperVCF">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.somaticsniperVCF</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF as produced by somaticsniper.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.threads">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of threads to use.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.timeMinutes">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.timeMinutes</b><br />
        <i>Int &mdash; Default: 60</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.varscanIndel">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.varscanIndel</b><br />
        <i>File? &mdash; Default: None</i><br />
        An indel VCF as produced by varscan.
</p>
<p name="Somatic.somaticVariantcalling.pairedSomaticSeq.varscanSNV">
        <b>Somatic.somaticVariantcalling.pairedSomaticSeq.varscanSNV</b><br />
        <i>File? &mdash; Default: None</i><br />
        An SNV VCF as produced by varscan.
</p>
<p name="Somatic.somaticVariantcalling.pairedTraining.threads">
        <b>Somatic.somaticVariantcalling.pairedTraining.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of threads to use.
</p>
<p name="Somatic.somaticVariantcalling.pairedTraining.timeMinutes">
        <b>Somatic.somaticVariantcalling.pairedTraining.timeMinutes</b><br />
        <i>Int &mdash; Default: 240</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.singleSomaticSeq.lofreqVCF">
        <b>Somatic.somaticVariantcalling.singleSomaticSeq.lofreqVCF</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF as produced by lofreq.
</p>
<p name="Somatic.somaticVariantcalling.singleSomaticSeq.scalpelVCF">
        <b>Somatic.somaticVariantcalling.singleSomaticSeq.scalpelVCF</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF as produced by scalpel.
</p>
<p name="Somatic.somaticVariantcalling.singleSomaticSeq.threads">
        <b>Somatic.somaticVariantcalling.singleSomaticSeq.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of threads to use.
</p>
<p name="Somatic.somaticVariantcalling.singleSomaticSeq.timeMinutes">
        <b>Somatic.somaticVariantcalling.singleSomaticSeq.timeMinutes</b><br />
        <i>Int &mdash; Default: 60</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.singleSomaticSeq.varscanVCF">
        <b>Somatic.somaticVariantcalling.singleSomaticSeq.varscanVCF</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF as produced by varscan.
</p>
<p name="Somatic.somaticVariantcalling.singleTraining.threads">
        <b>Somatic.somaticVariantcalling.singleTraining.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of threads to use.
</p>
<p name="Somatic.somaticVariantcalling.singleTraining.timeMinutes">
        <b>Somatic.somaticVariantcalling.singleTraining.timeMinutes</b><br />
        <i>Int &mdash; Default: 240</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.sitesForContamination">
        <b>Somatic.somaticVariantcalling.sitesForContamination</b><br />
        <i>File? &mdash; Default: None</i><br />
        A bed file, vcf file or interval list with regions for GetPileupSummaries to operate on.
</p>
<p name="Somatic.somaticVariantcalling.sitesForContaminationIndex">
        <b>Somatic.somaticVariantcalling.sitesForContaminationIndex</b><br />
        <i>File? &mdash; Default: None</i><br />
        The index for the vcf file provided to sitesForContamination.
</p>
<p name="Somatic.somaticVariantcalling.snvIndex.timeMinutes">
        <b>Somatic.somaticVariantcalling.snvIndex.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputFile,"G"))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.addGTFieldIndels.outputVCFName">
        <b>Somatic.somaticVariantcalling.strelka.addGTFieldIndels.outputVCFName</b><br />
        <i>String &mdash; Default: basename(strelkaVCF,".gz")</i><br />
        The location the output VCF file should be written to.
</p>
<p name="Somatic.somaticVariantcalling.strelka.addGTFieldIndels.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.addGTFieldIndels.timeMinutes</b><br />
        <i>Int &mdash; Default: 20</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.addGTFieldSVs.outputVCFName">
        <b>Somatic.somaticVariantcalling.strelka.addGTFieldSVs.outputVCFName</b><br />
        <i>String &mdash; Default: basename(strelkaVCF,".gz")</i><br />
        The location the output VCF file should be written to.
</p>
<p name="Somatic.somaticVariantcalling.strelka.addGTFieldSVs.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.addGTFieldSVs.timeMinutes</b><br />
        <i>Int &mdash; Default: 20</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.addGTFieldVariants.outputVCFName">
        <b>Somatic.somaticVariantcalling.strelka.addGTFieldVariants.outputVCFName</b><br />
        <i>String &mdash; Default: basename(strelkaVCF,".gz")</i><br />
        The location the output VCF file should be written to.
</p>
<p name="Somatic.somaticVariantcalling.strelka.addGTFieldVariants.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.addGTFieldVariants.timeMinutes</b><br />
        <i>Int &mdash; Default: 20</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.bedPrepare.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.bedPrepare.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputFile,"G"))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.combineVariants.dockerImage">
        <b>Somatic.somaticVariantcalling.strelka.combineVariants.dockerImage</b><br />
        <i>String &mdash; Default: "broadinstitute/gatk3:3.8-1"</i><br />
        The docker image used for this task. Changing this may result in errors which the developers may choose not to address.
</p>
<p name="Somatic.somaticVariantcalling.strelka.combineVariants.filteredRecordsMergeType">
        <b>Somatic.somaticVariantcalling.strelka.combineVariants.filteredRecordsMergeType</b><br />
        <i>String &mdash; Default: "KEEP_IF_ANY_UNFILTERED"</i><br />
        Equivalent to CombineVariants' `--filteredrecordsmergetype` option.
</p>
<p name="Somatic.somaticVariantcalling.strelka.combineVariants.genotypeMergeOption">
        <b>Somatic.somaticVariantcalling.strelka.combineVariants.genotypeMergeOption</b><br />
        <i>String &mdash; Default: "UNIQUIFY"</i><br />
        Equivalent to CombineVariants' `--genotypemergeoption` option.
</p>
<p name="Somatic.somaticVariantcalling.strelka.combineVariants.javaXmx">
        <b>Somatic.somaticVariantcalling.strelka.combineVariants.javaXmx</b><br />
        <i>String &mdash; Default: "12G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.strelka.combineVariants.memory">
        <b>Somatic.somaticVariantcalling.strelka.combineVariants.memory</b><br />
        <i>String &mdash; Default: "13G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.combineVariants.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.combineVariants.timeMinutes</b><br />
        <i>Int &mdash; Default: 180</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherIndels.compressionLevel">
        <b>Somatic.somaticVariantcalling.strelka.gatherIndels.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The compression level at which the BAM files are written.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherIndels.javaXmx">
        <b>Somatic.somaticVariantcalling.strelka.gatherIndels.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherIndels.memory">
        <b>Somatic.somaticVariantcalling.strelka.gatherIndels.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherIndels.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.gatherIndels.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputVCFs,"G")) * 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherIndels.useJdkDeflater">
        <b>Somatic.somaticVariantcalling.strelka.gatherIndels.useJdkDeflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java deflator to compress the BAM files. False uses the optimized intel deflater.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherIndels.useJdkInflater">
        <b>Somatic.somaticVariantcalling.strelka.gatherIndels.useJdkInflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java inflater. False, uses the optimized intel inflater.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherSVs.compressionLevel">
        <b>Somatic.somaticVariantcalling.strelka.gatherSVs.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The compression level at which the BAM files are written.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherSVs.javaXmx">
        <b>Somatic.somaticVariantcalling.strelka.gatherSVs.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherSVs.memory">
        <b>Somatic.somaticVariantcalling.strelka.gatherSVs.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherSVs.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.gatherSVs.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputVCFs,"G")) * 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherSVs.useJdkDeflater">
        <b>Somatic.somaticVariantcalling.strelka.gatherSVs.useJdkDeflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java deflator to compress the BAM files. False uses the optimized intel deflater.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherSVs.useJdkInflater">
        <b>Somatic.somaticVariantcalling.strelka.gatherSVs.useJdkInflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java inflater. False, uses the optimized intel inflater.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherVariants.compressionLevel">
        <b>Somatic.somaticVariantcalling.strelka.gatherVariants.compressionLevel</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The compression level at which the BAM files are written.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherVariants.javaXmx">
        <b>Somatic.somaticVariantcalling.strelka.gatherVariants.javaXmx</b><br />
        <i>String &mdash; Default: "4G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherVariants.memory">
        <b>Somatic.somaticVariantcalling.strelka.gatherVariants.memory</b><br />
        <i>String &mdash; Default: "5G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherVariants.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.gatherVariants.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputVCFs,"G")) * 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherVariants.useJdkDeflater">
        <b>Somatic.somaticVariantcalling.strelka.gatherVariants.useJdkDeflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java deflator to compress the BAM files. False uses the optimized intel deflater.
</p>
<p name="Somatic.somaticVariantcalling.strelka.gatherVariants.useJdkInflater">
        <b>Somatic.somaticVariantcalling.strelka.gatherVariants.useJdkInflater</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        True, uses the java inflater. False, uses the optimized intel inflater.
</p>
<p name="Somatic.somaticVariantcalling.strelka.indelsIndex.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.indelsIndex.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputFile,"G"))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.mantaSomatic.cores">
        <b>Somatic.somaticVariantcalling.strelka.mantaSomatic.cores</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of cores to use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.mantaSomatic.memoryGb">
        <b>Somatic.somaticVariantcalling.strelka.mantaSomatic.memoryGb</b><br />
        <i>Int &mdash; Default: 4</i><br />
        The amount of memory this job will use in Gigabytes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.mantaSomatic.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.mantaSomatic.timeMinutes</b><br />
        <i>Int &mdash; Default: 60</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.scatterList.memory">
        <b>Somatic.somaticVariantcalling.strelka.scatterList.memory</b><br />
        <i>String &mdash; Default: "256M"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.scatterList.prefix">
        <b>Somatic.somaticVariantcalling.strelka.scatterList.prefix</b><br />
        <i>String &mdash; Default: "scatters/scatter-"</i><br />
        The prefix of the ouput files. Output will be named like: <PREFIX><N>.bed, in which N is an incrementing number. Default 'scatter-'.
</p>
<p name="Somatic.somaticVariantcalling.strelka.scatterList.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.scatterList.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.scatterSize">
        <b>Somatic.somaticVariantcalling.strelka.scatterSize</b><br />
        <i>Int? &mdash; Default: None</i><br />
        The size of the scattered regions in bases. Scattering is used to speed up certain processes. The genome will be seperated into multiple chunks (scatters) which will be processed in their own job, allowing for parallel processing. Higher values will result in a lower number of jobs. The optimal value here will depend on the available resources.
</p>
<p name="Somatic.somaticVariantcalling.strelka.scatterSizeMillions">
        <b>Somatic.somaticVariantcalling.strelka.scatterSizeMillions</b><br />
        <i>Int &mdash; Default: 1000</i><br />
        Same as scatterSize, but is multiplied by 1000000 to get scatterSize. This allows for setting larger values more easily
</p>
<p name="Somatic.somaticVariantcalling.strelka.strelkaGermline.cores">
        <b>Somatic.somaticVariantcalling.strelka.strelkaGermline.cores</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of cores to use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.strelkaGermline.memoryGb">
        <b>Somatic.somaticVariantcalling.strelka.strelkaGermline.memoryGb</b><br />
        <i>Int &mdash; Default: 4</i><br />
        The amount of memory this job will use in Gigabytes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.strelkaGermline.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.strelkaGermline.timeMinutes</b><br />
        <i>Int &mdash; Default: 90</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.strelkaSomatic.cores">
        <b>Somatic.somaticVariantcalling.strelka.strelkaSomatic.cores</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of cores to use.
</p>
<p name="Somatic.somaticVariantcalling.strelka.strelkaSomatic.memoryGb">
        <b>Somatic.somaticVariantcalling.strelka.strelkaSomatic.memoryGb</b><br />
        <i>Int &mdash; Default: 4</i><br />
        The amount of memory this job will use in Gigabytes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.strelkaSomatic.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.strelkaSomatic.timeMinutes</b><br />
        <i>Int &mdash; Default: 90</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.svsIndex.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.svsIndex.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputFile,"G"))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.strelka.variantsIndex.timeMinutes">
        <b>Somatic.somaticVariantcalling.strelka.variantsIndex.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil(size(inputFile,"G"))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.trainingSet">
        <b>Somatic.somaticVariantcalling.trainingSet</b><br />
        <i>struct(jsmVCF : File?, lofreqIndel : File?, lofreqSNV : File?, museVCF : File?, mutect2VCF : File?, normalBam : File?, normalBamIndex : File?, scalpelVCF : File?, somaticsniperVCF : File?, strelkaIndel : File?, strelkaSNV : File?, truthIndel : File, truthSNV : File, tumorBam : File, tumorBamIndex : File, vardictVCF : File?, varscanIndel : File?, varscanSNV : File?)? &mdash; Default: None</i><br />
        VCF files used to train somaticseq.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryAlignments">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryAlignments</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not supplementary reads should be filtered before vardict is run.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.excludeSpecificFilter">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.excludeSpecificFilter</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to samtools view's `-G` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.includeFilter">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.includeFilter</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to samtools view's `-f` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.MAPQthreshold">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.MAPQthreshold</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to samtools view's `-q` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.memory">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.memory</b><br />
        <i>String &mdash; Default: "1G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.referenceFasta">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.referenceFasta</b><br />
        <i>File? &mdash; Default: None</i><br />
        The reference fasta file also used for mapping.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.threads">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of threads to use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.timeMinutes">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inFile,"G") * 5))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.uncompressedBamOutput">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryControl.uncompressedBamOutput</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to samtools view's `-u` flag.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.excludeSpecificFilter">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.excludeSpecificFilter</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to samtools view's `-G` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.includeFilter">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.includeFilter</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to samtools view's `-f` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.MAPQthreshold">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.MAPQthreshold</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to samtools view's `-q` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.memory">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.memory</b><br />
        <i>String &mdash; Default: "1G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.referenceFasta">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.referenceFasta</b><br />
        <i>File? &mdash; Default: None</i><br />
        The reference fasta file also used for mapping.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.threads">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of threads to use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.timeMinutes">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(inFile,"G") * 5))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.uncompressedBamOutput">
        <b>Somatic.somaticVariantcalling.vardict.filterSupplementaryTumor.uncompressedBamOutput</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Equivalent to samtools view's `-u` flag.
</p>
<p name="Somatic.somaticVariantcalling.vardict.gatherVcfs.javaXmx">
        <b>Somatic.somaticVariantcalling.vardict.gatherVcfs.javaXmx</b><br />
        <i>String &mdash; Default: "8G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.vardict.gatherVcfs.memory">
        <b>Somatic.somaticVariantcalling.vardict.gatherVcfs.memory</b><br />
        <i>String &mdash; Default: "9G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.gatherVcfs.timeMinutes">
        <b>Somatic.somaticVariantcalling.vardict.gatherVcfs.timeMinutes</b><br />
        <i>Int &mdash; Default: 1 + ceil((size(vcfFiles,"G") * 5))</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.vardict.scatterList.chunkSize">
        <b>Somatic.somaticVariantcalling.vardict.scatterList.chunkSize</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to chunked-scatter's `-c` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.scatterList.memory">
        <b>Somatic.somaticVariantcalling.vardict.scatterList.memory</b><br />
        <i>String &mdash; Default: "256M"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.scatterList.minimumBasesPerFile">
        <b>Somatic.somaticVariantcalling.vardict.scatterList.minimumBasesPerFile</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to chunked-scatter's `-m` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.scatterList.overlap">
        <b>Somatic.somaticVariantcalling.vardict.scatterList.overlap</b><br />
        <i>Int? &mdash; Default: None</i><br />
        Equivalent to chunked-scatter's `-o` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.scatterList.prefix">
        <b>Somatic.somaticVariantcalling.vardict.scatterList.prefix</b><br />
        <i>String &mdash; Default: "./scatter"</i><br />
        The prefix for the output files.
</p>
<p name="Somatic.somaticVariantcalling.vardict.scatterList.splitContigs">
        <b>Somatic.somaticVariantcalling.vardict.scatterList.splitContigs</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        If set, contigs are allowed to be split up over multiple files.
</p>
<p name="Somatic.somaticVariantcalling.vardict.scatterList.timeMinutes">
        <b>Somatic.somaticVariantcalling.vardict.scatterList.timeMinutes</b><br />
        <i>Int &mdash; Default: 2</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.chromosomeColumn">
        <b>Somatic.somaticVariantcalling.vardict.varDict.chromosomeColumn</b><br />
        <i>Int &mdash; Default: 1</i><br />
        Equivalent to vardict-java's `-c` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.endColumn">
        <b>Somatic.somaticVariantcalling.vardict.varDict.endColumn</b><br />
        <i>Int &mdash; Default: 3</i><br />
        Equivalent to vardict-java's `-E` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.geneColumn">
        <b>Somatic.somaticVariantcalling.vardict.varDict.geneColumn</b><br />
        <i>Int &mdash; Default: 4</i><br />
        Equivalent to vardict-java's `-g` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.javaXmx">
        <b>Somatic.somaticVariantcalling.vardict.varDict.javaXmx</b><br />
        <i>String &mdash; Default: "16G"</i><br />
        The maximum memory available to the program. Should be lower than `memory` to accommodate JVM overhead.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.mappingQuality">
        <b>Somatic.somaticVariantcalling.vardict.varDict.mappingQuality</b><br />
        <i>Float &mdash; Default: 20</i><br />
        Equivalent to var2vcf_paired.pl or var2vcf_valid.pl's `-Q` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.memory">
        <b>Somatic.somaticVariantcalling.vardict.varDict.memory</b><br />
        <i>String &mdash; Default: "18G"</i><br />
        The amount of memory this job will use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.minimumAlleleFrequency">
        <b>Somatic.somaticVariantcalling.vardict.varDict.minimumAlleleFrequency</b><br />
        <i>Float &mdash; Default: 0.02</i><br />
        Equivalent to var2vcf_paired.pl or var2vcf_valid.pl's `-f` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.minimumTotalDepth">
        <b>Somatic.somaticVariantcalling.vardict.varDict.minimumTotalDepth</b><br />
        <i>Int &mdash; Default: 8</i><br />
        Equivalent to var2vcf_paired.pl or var2vcf_valid.pl's `-d` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.minimumVariantDepth">
        <b>Somatic.somaticVariantcalling.vardict.varDict.minimumVariantDepth</b><br />
        <i>Int &mdash; Default: 4</i><br />
        Equivalent to var2vcf_paired.pl or var2vcf_valid.pl's `-v` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.outputAllVariantsAtSamePosition">
        <b>Somatic.somaticVariantcalling.vardict.varDict.outputAllVariantsAtSamePosition</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to var2vcf_paired.pl or var2vcf_valid.pl's `-A` flag.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.outputCandidateSomaticOnly">
        <b>Somatic.somaticVariantcalling.vardict.varDict.outputCandidateSomaticOnly</b><br />
        <i>Boolean &mdash; Default: true</i><br />
        Equivalent to var2vcf_paired.pl or var2vcf_valid.pl's `-M` flag.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.startColumn">
        <b>Somatic.somaticVariantcalling.vardict.varDict.startColumn</b><br />
        <i>Int &mdash; Default: 2</i><br />
        Equivalent to vardict-java's `-S` option.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.threads">
        <b>Somatic.somaticVariantcalling.vardict.varDict.threads</b><br />
        <i>Int &mdash; Default: 1</i><br />
        The number of threads to use.
</p>
<p name="Somatic.somaticVariantcalling.vardict.varDict.timeMinutes">
        <b>Somatic.somaticVariantcalling.vardict.varDict.timeMinutes</b><br />
        <i>Int &mdash; Default: 300</i><br />
        The maximum amount of time the job will run in minutes.
</p>
<p name="Somatic.somaticVariantcalling.variantsForContamination">
        <b>Somatic.somaticVariantcalling.variantsForContamination</b><br />
        <i>File? &mdash; Default: None</i><br />
        A VCF file with common variants.
</p>
<p name="Somatic.somaticVariantcalling.variantsForContaminationIndex">
        <b>Somatic.somaticVariantcalling.variantsForContaminationIndex</b><br />
        <i>File? &mdash; Default: None</i><br />
        The index of the common variants VCF file.
</p>
<p name="Somatic.useBwaKit">
        <b>Somatic.useBwaKit</b><br />
        <i>Boolean &mdash; Default: false</i><br />
        Whether or not BWA kit should be used. If false BWA mem will be used.
</p>
</details>








<hr />

> Generated using WDL AID (0.1.1)
